app/services/password_crypt.rb

```diff
@@ -7,9 +7,14 @@ class PasswordCrypt
     ALGORITHMS['MD5'] = '$1$'
   end
 
+  def self.generate_salt
+    # Linux crypt accepts maximum 16 [a-zA-Z0-9./] characters
+    SecureRandom.base64(12).tr('+=', '..')
```

Comment:
wouldn't this make the distribution of characters in the salt unrandom?

Reply:
After small research, it looks like the most important aspect of salt is uniqueness and not randomness. Since I am bumping the salt length from 6 to 12 characters and given the complexity and cost of maintenance of some properly distributed salt generator like https://gist.github.com/kennethkalmer/104980 I think it would be reasonable to waive this.

Unless there is some nice way of generating base63 random string. What you think?

Reply:
How about just using `SecureRandom.alphanumeric` instead?

Reply:
Good idea, we are loosing two characters but on the other hand I can make it longer up to the maximum of 16 characters now.